File: prog3/trabalho/src/trabalho/Trabalho.java
package trabalho;

import java.io.*;
import java.util.*;
/**
 * <i>Documentação da classe Trabalho </i>
 *
 * @version 1.0
 * @since 05/05/2016
 * @author Jordão e Nicole
 */
public class Trabalho extends Avaliacao implements Serializable {

    private final int tamanho;
    private int qtdGrupos;

    public Trabalho(Disciplina d, String codigo, String nome, double peso, String tipo, int tamanho, Date data) {
        super(d, codigo, nome, peso, tipo, data);
        this.tamanho = tamanho;
        qtdGrupos = 0;
    }

    int getQuantidadeDeGrupos() {
        return qtdGrupos;
    }

    void setQuantidadeDeGrupos(int n) {
        qtdGrupos = n;
    }

    public int getTamanho() {
        return tamanho;
    }

    @Override
    public String toString() {
        return super.toString() + ", tamanho máximo de alunos por grupo: " + tamanho;
    }
}

File: prog3/trabalho2/ufes_sys/sys/Trabalho.cpp
/* 
 * File:   Trabalho.cpp
 * Author: nicole e jordao
 * 
 * Created on 17 de Junho de 2016, 10:00
 */

#include "Trabalho.hpp"
namespace ufes_sys {

    Trabalho::Trabalho(Disciplina *disciplina, string codigo, string nome, string data, string tipo, double peso,int tamanho_grupo)
            : Avaliacao(disciplina, codigo, nome, data, tipo, peso) {
        this->tamanho_grupo = tamanho_grupo;
    }


    Trabalho::~Trabalho() {
    }
}
